patches/net/minecraft/client/data/models/MultiVariant.java.patch

```diff
@@ -1,45 +1,54 @@
 --- a/net/minecraft/client/data/models/MultiVariant.java
 +++ b/net/minecraft/client/data/models/MultiVariant.java
-@@ -12,20 +_,41 @@
+@@ -12,20 +_,50 @@
  import net.neoforged.api.distmarker.OnlyIn;
  
  @OnlyIn(Dist.CLIENT)
 -public record MultiVariant(WeightedList<Variant> variants) {
 -    public MultiVariant(WeightedList<Variant> variants) {
-+public record MultiVariant(WeightedList<Variant> variants, @org.jetbrains.annotations.Nullable net.neoforged.neoforge.client.model.generators.blockstate.CustomBlockStateModelBuilder customBlockStateModel) {
++public record MultiVariant(WeightedList<Variant> variants, WeightedList<net.neoforged.neoforge.client.model.generators.blockstate.CustomBlockStateModelBuilder> customBlockStateModels) {
 +    /**
-+     * @param customBlockStateModel Neo-added parameter to allow using custom blockstate models with MultiVariantGenerator and MultiPartGenerator (only used during datagen)
++     * @param customBlockStateModels Neo-added parameter to allow using custom blockstate models with MultiVariantGenerator and MultiPartGenerator (only used during datagen)
 +     */
-+    public MultiVariant(WeightedList<Variant> variants, @org.jetbrains.annotations.Nullable net.neoforged.neoforge.client.model.generators.blockstate.CustomBlockStateModelBuilder customBlockStateModel) {
-+        if (customBlockStateModel != null) {
++    public MultiVariant(WeightedList<Variant> variants, WeightedList<net.neoforged.neoforge.client.model.generators.blockstate.CustomBlockStateModelBuilder> customBlockStateModels) {
++        if (!customBlockStateModels.isEmpty()) {
 +            if (!variants.isEmpty()) {
 +                throw new IllegalArgumentException("Cannot specify both a variant list and a custom blockstate model");
 +            }
 +            this.variants = variants;
-+            this.customBlockStateModel = customBlockStateModel;
++            this.customBlockStateModels = customBlockStateModels;
 +        } else
          if (variants.isEmpty()) {
              throw new IllegalArgumentException("Variant list must contain at least one element");
          } else {
              this.variants = variants;
-+            this.customBlockStateModel = null;
++            this.customBlockStateModels = customBlockStateModels;
          }
 +    }
 +
 +    public MultiVariant(WeightedList<Variant> variants) {
-+        this(variants, null);
++        this(variants, WeightedList.of());
++    }
++
++    // Neo: convenience function for datagen with custom blockstate models
++    public static MultiVariant of(WeightedList<net.neoforged.neoforge.client.model.generators.blockstate.CustomBlockStateModelBuilder> customBlockStateModels) {
```

Comment:
There should still be a function that accepts a single `CustomBlockStateModelBuilder`, no?

Reply:
Sure, why not